lib/devise/failure_app.rb

```diff
@@ -51,7 +51,18 @@ def http_auth
     end
 
     def recall
-      env["PATH_INFO"]  = attempted_path
+      config = Rails.application.config
+
+      if config.try(:relative_url_root)
+        base_path = Pathname.new(config.relative_url_root)
```

Comment:
The method makes use of Pathname's `relative_path_from` method. Is there a better substitute? As far as I can tell, there isn't anything else.